Rubberduck.CodeAnalysis/QuickFixes/IgnoreOnceQuickFix.cs

```diff
@@ -29,98 +31,109 @@ public IgnoreOnceQuickFix(RubberduckParserState state, IEnumerable<IInspection>
-            var component = _state.ProjectsProvider.Component(result.QualifiedSelection.QualifiedName);
-            using (var module = component.CodeModule)
+            if (result.Target?.DeclarationType.HasFlag(DeclarationType.Module) == true)
             {
-                annotationLine = result.QualifiedSelection.Selection.StartLine;
-                while (annotationLine != 1 && module.GetLines(annotationLine - 1, 1).EndsWith(" _"))
-                {
-                    annotationLine--;
-                }
+                FixModule(result, rewriteSession);
             }
-
-            RuleContext treeRoot = result.Context;
-            while (treeRoot.Parent != null)
+            else
             {
-                treeRoot = treeRoot.Parent;
+                FixNonModule(result, rewriteSession);
             }
+        }
 
-            var listener = new CommentOrAnnotationListener();
-            ParseTreeWalker.Default.Walk(listener, treeRoot);
-            var commentContext = listener.Contexts.LastOrDefault(i => i.Stop.TokenIndex <= result.Context.Start.TokenIndex);
-            var commented = commentContext?.Stop.Line + 1 == annotationLine;
+        private void FixNonModule(IInspectionResult result, IRewriteSession rewriteSession)
+        {
+            int insertionIndex;
+            string insertText;
+            var annotationText = $"'@Ignore {result.Inspection.AnnotationName}";
 
-            var rewriter = rewriteSession.CheckOutModuleRewriter(result.QualifiedSelection.QualifiedName);
+            var module = result.QualifiedSelection.QualifiedName;
+            var parseTree = _state.GetParseTree(module, CodeKind.CodePaneCode);
+            var eolListener = new EndOfLineListener();
+            ParseTreeWalker.Default.Walk(eolListener, parseTree);
+            var previousEol = eolListener.Contexts
+                .OrderBy(eol => eol.Start.TokenIndex)
+                .LastOrDefault(eol => eol.Start.Line < result.QualifiedSelection.Selection.StartLine);
 
-            if (commented)
+            var rewriter = rewriteSession.CheckOutModuleRewriter(module);
+
+            if (previousEol == null)
             {
-                var annotation = commentContext.annotationList()?.annotation(0);
-                if (annotation != null && annotation.GetText().StartsWith("Ignore"))
-                {
-                    rewriter.InsertAfter(annotation.annotationName().Stop.TokenIndex, $" {result.Inspection.AnnotationName},");
-                }
-                else
-                {
-                    var indent = new string(Enumerable.Repeat(' ', commentContext.Start.Column).ToArray());
-                    rewriter.InsertAfter(commentContext.Stop.TokenIndex, $"{indent}{annotationText}{Environment.NewLine}");
-                }
+                // The context to get annotated is on the first line; we need to insert before token index 0.
```

Comment:
We cannot determine this from the selection in general. The problem is that it might be on the first logical line but not the first physical line due to line continuations.

Because this is not performance critical code, I opted not to add a further case for when it is on the first physical line.